ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -312,8 +314,9 @@ private static PubsubMessageToObjectNode getFormat(Env env) {
       case "decoded":
         return PubsubMessageToObjectNode.Decoded.of();
       case "payload":
-        return PubsubMessageToObjectNode.Payload.of(env.getStrings(STRICT_SCHEMA_DOCTYPES, null),
-            env.getString(SCHEMAS_LOCATION), FileInputStream::new);
+        return PubsubMessageToObjectNode.Payload.WithOpenCensusMetrics.of(
+            env.getStrings(STRICT_SCHEMA_DOCTYPES, null), env.getString(SCHEMAS_LOCATION),
+            FileInputStream::new);
```

Comment:
Would it be possible to amend the API so that this becomes:

```
return PubsubMessageToObjectNode.Payload
          .of(env.getStrings(STRICT_SCHEMA_DOCTYPES, null),
              env.getString(SCHEMAS_LOCATION), FileInputStream::new)
          .withOpenCensusMetrics();
```

So we'd construct a `Payload`, and then the `withOpenCensusMetrics()` method would return a `WithOpenCensusMetrics` with the relevant fields copied from the base `Payload` object.

Reply:
that sounds excellent, will amend